File: easy/ts/reverse_bits.ts
function reverseBits(n: number): number {
  let result = 0;
  let count = 0;
  while (count < 32) {
    result <<= 1;
    if (n & 1) result ^= 1;
    n >>= 1;
    count++;
  }

  return result >>>= 0;
};

File: easy/python/reverse_bits.py
class Solution:
    def reverseBits(self, n: int) -> int:
        res = 0
        count = 0
        while count < 32:
            res <<= 1
            if n & 1:
                res ^= 1
            n >>= 1
            count += 1
        return res
